Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityDeviceManager.cs

```diff
@@ -653,6 +653,8 @@ private BaseWindowsMixedRealitySource GetOrAddController(InteractionSource inter
                         // Return null so we don't raise the source detected.
                         return null;
                     }
+                    else
+                        ((WindowsMixedRealityController)detectedController).EnsureControllerModel(interactionSource);
```

Comment:
Since loading the model from the API at runtime is an async call and might span multiple frames, is there the potential that we'll either:

1. Block here for a while while that loads?
or
2. Still return before the model has been loaded?

Reply:
Now that you mention it.
In my opinion, we should block until the model is loaded. I think it would make for a bad user experience waving a linerenderer and cursor without seeing the controller.
This should probably be part of the BaseController as it's the class with the Visualizer implementation.

Reply:
I think that makes sense. We'll just need to make sure it accounts for the case where a controller was connected then lost before the model finished loading.

Reply:
Ok, I implemented that. I moved the EnsureControllerModel call towards the detection stage, because previously it was part of the Update loop of the controller, which feels unnecessary. What I did not understand though, was the comment on the WindowsMixedRealityController's TryRenderControllerModel method:
>             // Note: Obtaining models from the driver will require access to the InteractionSource.
            // It's unclear whether the interaction source will be available during setup, so we attempt to create
            // the controller model on an input update

Reply:
Yeah, I'm not super sure about that comment either. I think it's a side-effect of overriding the base `TryRenderControllerModel`, which can't have platform-specific references (like the interaction source).
It does seem like something we can pass in though, similar to what you're doing in this PR.

Reply:
My primary concern (and perhaps @keveleigh's) is that blocking until the model loads would create a noticeable glitch in the app.

Is there any way to avoid blocking?

Reply:
won't block, this code path doesn't exist anymore. Only place the controller is loaded is in GetOrAddController of the WindowsMixedRealityDeviceManager.